### DISCUSSION TITLE
Run change event on input field with wire:model.defer

### DISCUSSION THREAD

**Discussion by mccarlosen** [livewire/livewire#5513] [1. Help] [answered]
I have the next code:

```html
<!-- blade -->
<input type="text" wire:model.defer="form.title" />
```

I have an event to capture the user's input with `keypress` and transform it to uppercase. And I use `e.preventDefault()` to prevent the default behavior.

```js
element.addEventListener('keypress', this.forceToUpperCase, false)

// ....

forceToUpperCase(e) {
    let charInput = e.keyCode;
    if ((charInput >= 97) && (charInput <= 122)) {
      if (!e.ctrlKey && !e.metaKey && !e.altKey) {
        let newChar = charInput - 32;
	newChar = String.fromCharCode(newChar).toUpperCase()
        let start = e.srcElement.selectionStart;
        let end = e.srcElement.selectionEnd;
        e.srcElement.value = e.srcElement.value.substring(0, start) + newChar + e.srcElement.value.substring(end);
        e.srcElement.setSelectionRange(start+1, start+1);
        if ("createEvent" in document) {
		let evt = new Event('change', {"bubbles": true, "cancelable": false});
		e.srcElement.dispatchEvent(evt);
	} else {
		e.srcElement.fireEvent("onchange");
	}
	e.preventDefault();
      }
   }
}
```
But this doesn't update the property of the form. I hope you can help me.

**Comment by joshhanley** [comment, marked as answer]
@mccarlosen yeah the issue is updating `.value` on an element using javascript won't update Livewire's value for the bound property.

So instead you need to call `@this.set('property', value, true)` and pass your property name, new value, and `true` (which will defer the change).

See here for details (it uses `$wire` as it's an Alpine example, but for vanilla javascript you use `@this`) https://laravel-livewire.com/docs/2.x/alpine-js#interacting-with-livewire-from-alpine

But personally I would write this with Alpine as the code is much simpler and it integrates nicely with Livewire.

Hope this helps!

**Reply by mccarlosen** [reply]
Great! Thank you for your help.

**Reply by joshhanley** [reply]
No worries! 🙂
